### DISCUSSION TITLE
Long-running Service method/RPC

### DISCUSSION THREAD

**Discussion by samsta** [eclipse-ecal/ecal#900] [Q&A]
Dear eCAL community,

In our system, we use both pub/sub and service methods a.k.a RPCs.

We have one particular RPC that can take up to 10 seconds to complete (it's not actually doing work, it's just waiting for something to happen), but when it does take that long, the process it is running in loses its subscriber information, i.e. any topic it publishes to from that point onwards does not actually make it to the subscribers.

In the `registration_timeout` in `ecal.ini` is set to 10000 (10 seconds), if I increase that number to something higher, the problem goes away, so I have a solution to my problem that works.

My question is, though, whether it's the right solution. It appears as though a long running RPC blocks some maintenance/monitoring tasks needed to maintain the registration info, and if that happens for long enough, the `registration_timeout` is hit. Do you have any guidance on long-running RPCs? Is there a way give the maintenance/monitoring tasks an explicit chance to run while the RPC is waiting, or is the official advice to either steer clear of long-running RPCs or make sure the `registration_timeout` is long enough?

Regards,
Sam

**Comment by FlorianReimold** [comment]
Thank you very much for reporting! We don't have a solution for you, but we already identified the cause for this issue and are going to fix it soon! I copied your description to Issue #903.

**Reply by rex-schilasky** [reply]
Issue fixed for upcoming release 5.11.0. Thanks again for reporting.

**Reply by samsta** [reply]
Thanks for getting onto it so quickly! Looking forward to 5.11.0
